src/ReinforcementLearningCore/src/core/hooks.jl

```diff
@@ -104,32 +104,44 @@ Base.getindex(h::RewardsPerEpisode) = h.rewards
 Store the total reward of each episode in the field of `rewards`. If
 `is_display_on_exit` is set to `true`, a unicode plot will be shown at the [`PostExperimentStage`](@ref).
 """
-Base.@kwdef mutable struct TotalRewardPerEpisode <: AbstractHook
-    rewards::Vector{Float64} = Float64[]
-    reward::Float64 = 0.0
-    is_display_on_exit::Bool = true
+mutable struct TotalRewardPerEpisode{T} <: AbstractHook where {T<:Union{Bool,Nothing}}
+    rewards::Vector{Float64}
+    reward::Float64
+    is_display_on_exit::Bool
+
+    function TotalRewardPerEpisode(; is_display_on_exit::Bool = true)
+        struct_type = is_display_on_exit ? Bool : Nothing
+        new{struct_type}(Float64[], 0.0, is_display_on_exit)
+    end
 end
 
 Base.getindex(h::TotalRewardPerEpisode) = h.rewards
 
-(h::TotalRewardPerEpisode)(::PostActStage, agent, env) = h.reward += reward(env)
+(h::TotalRewardPerEpisode)(::PostActStage, agent::AbstractPolicy, env::AbstractEnv) =
```

Comment:
Is this necessary for type stability ? Looks like this only restricts the type of agent and env.

Reply:
I'll drop this from this PR. But I think in a separate change, we should only define behavior on `agent::AbstractPolicy` and `env::AbstractEnv`; this may have performance relevance (I don't believe it does in this case), but from a user experience perspective it's helpful to see it in the code / function signature and should error if random other objects are used.